src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1161,9 +1159,9 @@ private HttpResponseHeaders CreateResponseHeaders(bool appCompleted)
         }
 
         // Set whether response can have body
-        _canWriteResponseBody = CanWriteResponseBody();
+        _responseBodyMode = CanWriteResponseBody() ? ResponseBodyMode.ContentLength : ResponseBodyMode.Disabled;
```

Comment:
This is why I suggested an `Unitialized` state. It seems wrong to have the mode be `ContentLength` both here and after calling `Reset()` when we could ultimately end up auto-chunking.